R/predict.R

```diff
@@ -0,0 +1,512 @@
+#' Predict Method for RtGam Models
+#'
+#' Generates predictions from an `RtGam` fit. Prediction dates can be specified
+#' flexibly using multiple approaches.
+#'
+#' @param object An `RtGam` object from [RtGam()].
+#' @param parameter A string specifying the prediction parameter. Options are
+#'   `"obs_cases"`, `"incidence"`, `"r"`, and `"Rt"`. Defaults to `"obs_cases"`.
+#' @param horizon Optional. Integer specifying forecast days from the last date
+#'   in the fit. For example, `horizon = 7` returns a 7-day forecast.
+#' @param min_date,max_date Optional. A date-like object. See details for more
+#'   information on specification.
+#' @param n Number of posterior samples to use. Default is 100.
+#' @param mean_delay Optional. Numeric mean delay used in the prediction.
+#' @param gi_pmf Optional. A vector representing the generation interval PMF.
+#' @param seed Random seed for reproducibility. Default is 12345.
+#' @param ... Additional arguments passed to lower-level functions.
+#'
+#' @details
+#' Prediction dates can be set in four ways:
+#'
+#' 1. **Using Fit Object Alone**: Predictions span the full date range in the
```

Comment:
what about supporting newdata pass in as often done? Currently you can't support covariates

Reply:
one option here could be to then make the other options actually be helper functions that adapt data to get to new data (i.e externally to predict). I don't feel strongly about this

Reply:
My first reaction is that it's a good point but should probably be handled together with https://github.com/CDCgov/cfa-gam-rt/issues/23. Thoughts on pushing this to that issue?

Reply:
yes agree